src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
> However, your argument works against TempStreamFactory just the same, e.g.:

and:

> We arrive again at the situation where someone can write one stream factory helper class that works for all implementations of PSR-7 supported by a single resource stream factory implementation

I disagree there, as the use case for a temp stream is not just to *open* it, but to *populate* it, the implementation of which you note in your helper example.

The problem I see with having a util package is that it's an *additional* dependency, and no longer becomes "depending on PSR-17", but "depending on PSR-17 and the PSR-17 helper library". I'd much rather my reusable code depend on the PSR-17 interfaces, in order to keep the dependencies minimal.

On top of that, there's then an extra onus on the proposed http-interop/util package: it would likely need to provide a factory for providing the above factory, and that gets into the issues that container-interop is delving into with its definition-interop subproject. Let's leave that to the implementations to tackle.

String content for messages is arguably the most common use case, which is an argument for the additional interface for temp streams as a first-class interface, despite it being a superset of resource streams from the implementation side.